### ISSUE TITLE
Possible code breaking change in guard/if case let statements

### ISSUE THREAD

**Issue by marciniwanicki** [nicklockwood/SwiftFormat#786]
I've noticed a small issue when formatting multiline `guard case` statements with associated types.

i.e.

```swift
enum IntOrString {
    case int(Int)
    case string(String)
}

let value = IntOrString.string("Test")

guard case IntOrString
     .string(let innerValue) else { // it might be a common format when the type name is long and enum case has multiple associated values
          return
}
```

the code above is formatted to:

```swift
guard case IntOrString
let .string(innerValue) = value else {
     return
}
```

Similar problem can be observed when formatting `if case let` statements.

Workaround:
- Remove the enum type name i.e. `guard let .string(innerValue) = value else {` 
- Put the enum type + case name in a single line i.e. `guard let IntOrString.string(innerValue) = value else {` 

Affected version:
0.47.1

**Comment by nicklockwood** [comment]
@marciniwanicki fixed in 0.47.2

**Comment by marciniwanicki** [comment]
Super! Many thanks @nicklockwood
